File: algorithm/HDU-ACM/ambition0109/1215.cpp
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: ambition0109
////Nickname: Ambition
////Run ID: 
////Submit time: 2010-04-28 11:02:28
////Compiler: Visual C
//////////////////////////////////////////////////////
////Problem ID: 1215
////Problem Title: 
////Run result: Accept
////Run time:62MS
////Run memory:2164KB
//////////////////System Comment End//////////////////
#include<stdio.h>
int num[500001];
main()
{
	int n,m,i=1,j;
	for(i=0;i<500001;i++) num[i]=1;
	for(i=2;i<250001;i++)
	{
		for(j=2;i*j<500001;j++)
			num[i*j]+=i;
	}
	for(scanf("%d",&n);n>0;n--)
	{
		scanf("%d",&m);
		printf("%d\n",num[m]);
	}
}

File: algorithm/杭电ACM/ambition0109/1215.java
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: ambition0109
////Nickname: Ambition
////Run ID: 
////Submit time: 2011-04-11 21:07:40
////Compiler: Java
//////////////////////////////////////////////////////
////Problem ID: 1215
////Problem Title: 
////Run result: Accept
////Run time:984MS
////Run memory:9192KB
//////////////////System Comment End//////////////////
import java.util.Arrays;
import java.util.Scanner;

public class Main {
	public static void main(String args[]){
		Scanner in=new Scanner(System.in);
		int[] num=new int[500005];
		Arrays.fill(num,0);
		for(int i=1;i<500000;i++){
			for(int j=i+i;j<=500000;j+=i){
				num[j]+=i;
			}
		}
		
		int t=in.nextInt();
		while(t--!=0){
			int n=in.nextInt();
			System.out.println(num[n]);
		}
	}
}
